plugins/async-node/ios/Sources/AsyncNodePlugin.swift

```diff
@@ -7,124 +7,149 @@
 
 import Foundation
 import JavaScriptCore
-
-#if SWIFT_PACKAGE
 import PlayerUI
-#endif
 
-public typealias AsyncHookHandler = (JSValue) async throws -> AsyncNodeHandlerType
+public typealias AsyncHookHandler = (JSValue, JSValue) async throws -> AsyncNodeHandlerType
 
 public enum AsyncNodeHandlerType {
     case multiNode([ReplacementNode])
     case singleNode(ReplacementNode)
+    case emptyNode
 }
 
 /**
  Wraps the core AsyncNodePlugin and taps into the `onAsyncNode` hook to allow asynchronous replacement of the node object that contains `async`
  */
 public class AsyncNodePlugin: JSBasePlugin, NativePlugin {
     public var hooks: AsyncNodeHook?
-
+    
     private var asyncHookHandler: AsyncHookHandler?
-
+    
     public var plugins: [JSBasePlugin] = []
-
+    
     /**
      Constructs the AsyncNodePlugin
      - Parameters:
-        - handler: The callback that is used to tap into the core `onAsyncNode` hook
-                   exposed to users of the plugin allowing them to supply the replacement node used in the tap callback
+     - handler: The callback that is used to tap into the core `onAsyncNode` hook
+     exposed to users of the plugin allowing them to supply the replacement node used in the tap callback
      */
     public convenience init(plugins: [JSBasePlugin] = [AsyncNodePluginPlugin()], _ handler: @escaping AsyncHookHandler) {
-
+        
         self.init(fileName: "AsyncNodePlugin.native", pluginName: "AsyncNodePlugin.AsyncNodePlugin")
         self.asyncHookHandler = handler
         self.plugins = plugins
     }
-
+    
+    /**
+     Converts a given `ReplacementNode` to a `JSValue` that can be used in the JavaScript context.
+     
+     - Parameters:
+     - replacementNode: The `ReplacementNode` to be converted.
+     - context: The `JSContext` in which the `JSValue` will be used.
+     - Returns: A `JSValue` representing the given `ReplacementNode`, or `nil` if the conversion fails.
+     */
+    func convertReplacementNodeToJSValue(_ replacementNode: ReplacementNode, context: JSContext) -> JSValue? {
```

Comment:
this can be moved to an extension of `ReplacementNode` to simplify and remove the first parameter

you can also call it directly to unit test it and force the `catch` block